boot-environ/src/environ/boot.clj

```diff
@@ -1,10 +1,45 @@
 (ns environ.boot
   {:boot/export-tasks true}
   (:require [boot.core :as core]
-            [environ.core :as environ]))
+            [environ.core :as environ]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]))
 
-(core/deftask environ [e env FOO=BAR {kw str} "The environment map"]
-  (fn [next-task]
-    (fn [fileset]
-      (with-redefs [environ/env (merge environ/env env)]
-        (next-task fileset)))))
+(def ^:const env-file ".boot-env")
+
+(defn- read-boot-env [fileset]
+  (if-let [t (core/tmp-get fileset env-file)]
+    (-> t
+        core/tmp-file
+        slurp
+        edn/read-string)
+    {}))
+
+(defn- write-boot-env [env tmp]
+  (let [s (prn-str env)
```

Comment:
Maybe do without the `let`? So something like:

```clojure
(spit (io/file tmp env-file) (prn-str env))
```

Again, a little more concise without losing readability.